### PR TITLE
Fix misspelling errors and typos reported by codespell

### LINKED ISSUE #10998: Codespell report for "OpenSSL" (on fossies.org)
The FOSS server [fossies.org](https://fossies.org) - also supporting "OpenSSL"  - offers a new feature "[Source code misspelling reports](https://fossies.org/features.html#codespell). Such reports are normally only generated on request, but as Fossies administrator I have just created for testing purposes such a [codespell](https://github.com/codespell-project/codespell) based analysis for the `openssl` release "1.1.1d":

 https://fossies.org/linux/misc/openssl/codespell.html

That version-independent (not linked) URL should always redirect to the last report (if available), so currently to

 https://fossies.org/linux/misc/openssl-1.1.1d.tar.gz/codespell.html

Although after a first review some obviously wrong matches ("False Positives" = FPs) are already filtered out (ignored) please inform me if you find more of them so that I can force a new improved check if applicable.

Just for information there are also two supplemental pages https://fossies.org/linux/misc/openssl/codespell_conf.html (showing some used "codespell" configurations) and https://fossies.org/linux/misc/openssl/codespell_fps.html (showing all resulting obvious "False Positives").

If  appropriate, additional reports for the current release or also master versions (these in a
special "test" folder) could be willingly created.

A remark: As a very new feature in the report page there was now added a one-letter column (also sortable), which should indicate the error context (mouse over optionally expands the letter information). This may make it easier to identify the potentially more serious errors, especially since additionally the letters for errors that are "only" in comments (`c` and `C`) and pure text files (`t`) are greyed out. The errors marked by a `?` may be in the source code itself. As C-layman I was unsure if the both spelling errors of that type `bion` (suggestion `biom`?) and `interm` (suggestion  `intern`?) are really spelling errors or FPs.

### PR DESCRIPTION
**Pull request by mspncp**
Fixes #10998

_(I'll try to cherry-pick to master if possible)_

### CODE REVIEW

**Comment by mspncp** [comment]
I just verified that the patch can be cherry-picked to master.

When merging, I'll go the opposite way, i.e., I push to master first and then cherry-pick back to 1.1.1, since that's our usual procedure.

@slontis please reconfirm for master.

**Comment by slontis** [comment]
reconfirmed.

**Comment by mspncp** [comment]
_(CHANGES and NEWS spelling errors fixed in #10545 as well, see 20d3e4e389dca22ce8e6da0818532cccb3f0b3f6.)_

**Comment by mspncp** [comment]
Sorry Shane, when logging off I just noticed that I had a lot of unsaved files in Emacs. So there is a lot more to review, see 3aa8f8cbe580d444598cc10b4f33f7a0ab2b61ac.

(I should go to bed now...)

**Comment by mspncp** [comment]
> I don't think these 'publicly' one's are right..

This brings us to an interesting discursion about grammar again. ;-)

Google seemse to have a clear preference towards [publicly](https://www.google.de/search?q=publicly) (262M) vs. [publically](https://www.google.de/search?q=publically) (8M). Also, many online dictionaries list the latter only as an alternative spelling. There is an interesting article about this subject on [grammarphobia.com](https://www.grammarphobia.com/blog/2015/08/publicly-publically.html), from where I took the following citation.

> The Merriam-Webster’s usage guide concludes its entry on “publically” with this advice: “You can use it if you like, but we do not really recommend it, because it will look unfamiliar to many who encounter it.”

I wonder what our british EU expatriate @mattcaswell thinks about it. But if you prefer that the previous spelling remains, I'll revert the changes.

**Comment by paulidale** [comment]
The Oxford lists it as _publicly_ noting that _publically_ is a misspelling:

> Note that the spelling is **publicly**, not **-ally**.

**Comment by mspncp** [comment]
> The Oxford lists it as _publicly_ noting that _publically_ is a misspelling:
> 
> > Note that the spelling is **publicly**, not **-ally**.

Yeah, I've seen that, too. I think it is listed either as uncommon or wrong, depending on how forgiving the dictionary is.

**Comment by slontis** [comment]
I hear a siren. I think the spelling police are coming.

There is also a conflict in rand_lcl.h

**Comment by DDvO** [comment]
There are certainly more important issues to worry about.

**Comment by mspncp** [comment]
> There are certainly more important issues to worry about than spelling variants.

Don't worry, we're just teasing eachother.

File: CHANGES [hunk 1]
```diff
@@ -8373,7 +8373,7 @@ des-cbc           3624.96k     5258.21k     5530.91k     5624.30k     5628.26k
 
   *) New OCSP utility. Allows OCSP requests to be generated or
      read. The request can be sent to a responder and the output
-     parsed, outputed or printed in text form. Not complete yet:
+     parsed, outputted or printed in text form. Not complete yet:
```
Comment (levitte) [inline CHANGES]: I was surprised to see this, but it seems that Merriam-Webster agrees with it, so...

**Comment by mattcaswell** [comment]
> I wonder what our british EU expatriate

Ok...no need to rub it in... :-(

> @mattcaswell thinks about it. But if you prefer that the previous spelling remains, I'll revert the changes.

publically looks wrong to me.

File: crypto/perlasm/README [hunk 2]
```diff
@@ -35,7 +35,7 @@ Functions of interest are:
 &function_begin_B(name,extra)	Same as normal function_begin but no pushing.
 &function_end(name)		Call at end of function.
 &function_end_A(name)		Standard pop and ret, for use inside functions
-&function_end_B(name)		Call at end but with poping or 'ret'.
+&function_end_B(name)		Call at end but with pop'ing or 'ret'.
```
Comment (mspncp) [inline crypto/perlasm/README]: > I didn't choose "popping" on purpose, to make the "pop" statement more apparent.

Revisiting this line, it does not seem like a real improvement to me. I will just change it to

```
Call at end but with 'pop' or 'ret'.
```

**Comment by mspncp** [comment]
> There is also a conflict in rand_lcl.h

The conflict was caused by the fact that I based my topic branch onto the tag OpenSSL_1_1_1d to get the same line numbers as in [codespell.openssl-1.1.1d.txt](https://fossies.org/linux/misc/openssl-1.1.1d.tar.gz/codespell.openssl-1.1.1d.txt) and forgot to rebase it to the tip of OpenSSL_1_1_1-stable afterwards. I just pushed a rebased version (without other changes).

Apropos rebase: git had no problems to detect the renaming `crypto/rand_rand_lcl.h` -> `crypto/rand_rand_local.h`. Maybe GitHub's rename limits are set too low.

```
$ git config -l | grep limit
merge.renamelimit=32768
diff.renamelimit=32768
```

**Comment by mspncp** [comment]
Merged to master and 1.1.1, thanks!

7fa8bcfe43 Fix misspelling errors and typos reported by codespell
0f68b771b0 Fix misspelling errors and typos reported by codespell

**Pull request closed without merging**